flink-metrics/flink-metrics-statsd/src/main/java/org/apache/flink/metrics/statsd/StatsDReporter.java

```diff
@@ -216,4 +238,14 @@ public String filterCharacters(String input) {
 
 		return chars == null ? input : new String(chars, 0, pos);
 	}
+
+	private boolean numberIsNegative(Number input) {
+		try {
+			return new BigDecimal(input.toString()).compareTo(BigDecimal.ZERO) == -1;
```

Comment:
I'm wondering whether we could use `return Double.compare(input.doubleValue(), 0) < 0` instead. This should be sufficient for our use-cases, and we get around the repeated convert-to-string -> parse-from-string operations.

Reply:
Yeah good call, in this case the rounding that doubleValue might do wouldn't matter anyhow. Changed it!